grafana/resource_dashboard.go

```diff
@@ -68,39 +86,32 @@ func ResourceDashboard() *schema.Resource {
 
 func CreateDashboard(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*client).gapi
-
-	dashboard := gapi.Dashboard{}
-
-	dashboard.Model = prepareDashboardModel(d.Get("config_json").(string))
-
-	dashboard.Folder = int64(d.Get("folder").(int))
-
-	dashboard.Overwrite = d.Get("overwrite").(bool)
-
+	dashboard := makeDashboard(d)
 	resp, err := client.NewDashboard(dashboard)
 	if err != nil {
 		return diag.FromErr(err)
 	}
-
-	d.SetId(resp.Slug)
-
+	d.SetId(resp.UID)
 	return ReadDashboard(ctx, d, meta)
 }
 
 func ReadDashboard(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*client).gapi
-
-	slug := d.Id()
-
-	dashboard, err := client.Dashboard(slug)
+	uid := d.Id()
```

Comment:
Won't the ID change make this fail when migrating from the old version to the new one? The ID stored in the state is the slug, it will try to get the dashboard by UID but with the slug and it will remove all dashboards from state since it won't find them

I think it might be nice to have an actual tfstate from a version before this one and use that to have a migration acceptance test

Reply:
Darn, you're right. I thought this seemed too easy. I'm able to reproduce this. I'm going to see if I can add a [StateUpgrader](https://www.terraform.io/docs/extend/resources/state-migration.html).

I'm not thrilled about committing raw state for acceptance tests. Seems to me we'll only need such tests to ensure _this_ PR works. I think I'll start by documenting in fine detail how I've tested once I get a StateUpgrader working. We can decide from there if it makes sense to commit something we should maintain beyond this.